packages/currency-utils/src/currencyUtils.ts

```diff
@@ -0,0 +1,32 @@
+import { Connection, PublicKey } from "@solana/web3.js";
+import { TOKEN_PROGRAM_ID, AccountLayout } from "@solana/spl-token";
+
+export const getBalance = async ({
+  pubKey,
+  mint,
+  connection,
+}: {
+  connection: Connection;
+  pubKey: PublicKey;
+  mint: string;
+}) => {
+  const tokenAccounts = await connection.getTokenAccountsByOwner(pubKey, {
```

Comment:
This is suboptimal. You should be able to use `getAssociatedTokenAddressSync` to get the address of the token account. Then fetch and decode that particular account with AccountLayout

Reply:
updated 🏁